serde_derive/src/internals/ast.rs

```diff
@@ -141,11 +143,16 @@ fn enum_from_ast<'a>(
     cx: &Ctxt,
     variants: &'a Punctuated<syn::Variant, Token![,]>,
     container_default: &attr::Default,
+    container_is_untagged: bool,
 ) -> Vec<Variant<'a>> {
     let variants: Vec<Variant> = variants
         .iter()
         .map(|variant| {
             let attrs = attr::Variant::from_ast(cx, variant);
+
+			if container_is_untagged && attrs.name().renamed() {
+				cx.error_spanned_by(&variant.ident, "renaming or adding a alias to a variant of an untagged enum does nothing");
+			}
```

Comment:
Ideally, error should point to the each problematic attribute instead of a variant name. Also, compile tests should be added to `test_suite/tests/ui`

Reply:
I fixed those two things. Are there any other attributes that need to have error messages like this?

Reply:
Thanks, much better, but it is better to point to the `alias = "..."` and `rename = "..."` parts only, so when you have multiple attributes on a single line, errors pointed to the corresponding parts of that line.

Don't known about other attributes. You may check other if they need improvements.

Reply:
I made it point towards the `alias` and `rename` parts to be even more clear. I think `#[serde(rename_all)]` may be another attribute that has this issue, but I couldn't get that error to work in 6d725642135e5d34fcde8cda00324b71d66b9e52.